### PR TITLE
OpenPaaS-Suite/esn#16 - Update Base Url

### LINKED ISSUE #16: Update docker-compose.yml to deploy with the new frontends
Currently, our [docker-compose.yml](https://github.com/linagora/openpaas-esn/blob/master/docker-compose.yml) deployment system is not using the new frontends. This issue objective is to have a deployment of OpenPaaS through this docker-compose.yml file, so that it also deploy/uses the new frontends.

Related issue: https://github.com/OpenPaaS-Suite/esn/issues/28

### PR DESCRIPTION
**Pull request by piavgh**
Partially resolves OpenPaaS-Suite/esn#16.

### CODE REVIEW

File: src/frontend/js/modules/http.js [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
   .provider('httpConfigurer', function() {
     var restangulars = [];
-    var baseUrl = '';
+    var baseUrl = window.openpaas && window.openpaas.BASE_URL ? window.openpaas.BASE_URL : '';
```
Comment (chamerling) [inline src/frontend/js/modules/http.js]: You should use the setBaseUrl function below which updates all the restangular instances. Do it from a `angular.run` block somewhere

File: src/frontend/js/modules/i18n/i18n-loader.service.js [hunk 2]
```diff
@@ -23,7 +23,8 @@
       };
 
       function getCatalogs() {
-        return $http.get('/api/i18n').then(function(res) {
+        const url = window.openpaas && window.openpaas.BASE_URL ? window.openpaas.BASE_URL + '/api/i18n' : '/api/i18n';
```
Comment (chamerling) [inline src/frontend/js/modules/i18n/i18n-loader.service.js]: Should be cleaner using http.js above

**Review by chamerling — DISMISSED** [review]

File: src/frontend/js/modules/http.js [hunk 3]
```diff
@@ -53,7 +53,7 @@
 
   .provider('httpConfigurer', function() {
     var restangulars = [];
-    var baseUrl = '';
+    var baseUrl = window.openpaas && window.openpaas.OPENPAAS_API_URL ? window.openpaas.OPENPAAS_API_URL : '';
```
Comment (tuanlc) [inline src/frontend/js/modules/http.js]:
```suggestion
    var baseUrl = window.openpaas && window.openpaas.OPENPAAS_API_URL || '';
```

File: src/frontend/js/modules/i18n/i18n-loader.service.js [hunk 4]
```diff
@@ -23,7 +23,8 @@
       };
 
       function getCatalogs() {
-        return $http.get('/api/i18n').then(function(res) {
+        const url = window.openpaas && window.openpaas.OPENPAAS_API_URL ? window.openpaas.OPENPAAS_API_URL + '/api/i18n' : '/api/i18n';
```
Comment (tuanlc) [inline src/frontend/js/modules/i18n/i18n-loader.service.js]:
```suggestion
        const url = (window.openpaas && window.openpaas.OPENPAAS_API_URL || '') + '/api/i18n';
```

Why this request does not use configured `BaseURL` above?

File: src/frontend/js/modules/i18n/i18n-loader.service.js (hunk 2 above)
Comment (piavgh) [reply src/frontend/js/modules/i18n/i18n-loader.service.js]: thanks. I updated it

File: src/frontend/js/modules/i18n/i18n-loader.service.js (hunk 4 above)
Comment (piavgh) [reply src/frontend/js/modules/i18n/i18n-loader.service.js]: thanks. I updated it

**Comment by itswillta** [comment]
Why is this still a draft?

**Review by tuanlc — DISMISSED** [review]

File: src/frontend/js/modules/i18n/i18n-loader.service.js [hunk 5]
```diff
@@ -23,7 +25,9 @@
       };
 
       function getCatalogs() {
-        return $http.get('/api/i18n').then(function(res) {
+        const url = httpConfigurer.getUrl('/api/i18n');
+
+        return $http.get(url).then(function(res) {
```
Comment (itswillta) [inline src/frontend/js/modules/i18n/i18n-loader.service.js:31]: You should add some comments to remind yourself or others to write tests for this part of the code later. e.g.:

```
// TODO: Write tests for this (https://github.com/OpenPaaS-Suite/esn-frontend-common-libs/pull/79)
```

Comment (piavgh) [reply src/frontend/js/modules/i18n/i18n-loader.service.js:31]: It's done. Please check again

**Review by itswillta — DISMISSED** [review]

**Pull request merged**